FXyz-Core/src/main/java/org/fxyz3d/shapes/primitives/TorusMesh.java

```diff
@@ -1,353 +1,341 @@
+ * names of its contributors may be used to endorse or promote products
+ * derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL F(X)yz BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */ 
+
+package org.fxyz3d.shapes.primitives;
+
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.scene.DepthTest;
+import javafx.scene.shape.MeshView;
+import javafx.scene.shape.TriangleMesh;
+
+import static java.lang.Math.cos;
+import static java.lang.Math.sin;
+
+/**
+ *
+ * @author jDub1581
+ */
+public class TorusMesh extends MeshView {
+
+    private static final int DEFAULT_DIVISIONS = 64;
+    private static final int DEFAULT_T_DIVISIONS = 64;
+    private static final double DEFAULT_RADIUS = 12.5;
+    private static final double DEFAULT_T_RADIUS = 5.0;
+    private static final double DEFAULT_START_ANGLE = 0.0;
+    private static final double DEFAULT_X_OFFSET = 0.0;
+    private static final double DEFAULT_Y_OFFSET = 0.0;
+    private static final double DEFAULT_Z_OFFSET = 1.0;
+
+    public TorusMesh() {
+        this(DEFAULT_DIVISIONS, DEFAULT_T_DIVISIONS, DEFAULT_RADIUS, DEFAULT_T_RADIUS);
+    }
+
+    public TorusMesh(double radius, double tubeRadius) {
+        this(DEFAULT_DIVISIONS, DEFAULT_T_DIVISIONS, radius, tubeRadius);
+    }
+
+    public TorusMesh(int radiusDivisions, int tubeDivisions, double radius, double tubeRadius) {
+        setRadiusDivisions(radiusDivisions);
+        setTubeDivisions(tubeDivisions);
+        setRadius(radius);
+        setTubeRadius(tubeRadius);
+        
+        setDepthTest(DepthTest.ENABLE);
+        updateMesh();
+    }
+
+    private void updateMesh(){       
+        setMesh(
+                createTorusMesh(
+                        getRadiusDivisions(),
+                        getTubeDivisions(),
+                        (float) getRadius(),
+                        (float) getTubeRadius(),
+                        (float) getTubeStartAngleOffset(),
+                        (float) getxOffset(),
+                        (float) getyOffset(),
+                        (float) getzOffset()
+                )
+        );
+    }
```

Comment:
Personally, I prefer the previous compact code style that avoids a lot of horizontal nesting, but I'm fine with this.